app/controllers/cost_types_controller.rb

```diff
@@ -90,12 +90,23 @@ def create
     flash.now[:error] = l(:notice_locking_conflict)
   end
 
-  def toggle_delete
-    @cost_type.deleted_at = @cost_type.deleted_at ?  nil : DateTime.now()
+  def destroy
+    @cost_type.deleted_at = DateTime.now
     @cost_type.default = false
 
     if @cost_type.save
-      flash[:notice] = @cost_type.deleted_at ? l(:notice_successful_delete) : l(:notice_successful_restore)
+      flash[:notice] = l(:notice_successful_delete)
+
+      redirect_back_or_default(:action => 'index')
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.